xbmc/epg/EpgContainer.cpp

```diff
@@ -379,10 +374,11 @@ void CEpgContainer::InsertFromDatabase(int iEpgID, const std::string &strName, c
   else
   {
     // create a new epg table
-    epg = new CEpg(iEpgID, strName, strScraperName, true);
+    epg = new CEpg(-1);
```

Comment:
why do we need to create a new instance here if we set epg to epgdb below.
dropping the this line and the check below should be fine.